File: v2/internal/runtime/js/runtime/window.js
/*
 _       __      _ __    
| |     / /___ _(_) /____
| | /| / / __ `/ / / ___/
| |/ |/ / /_/ / / (__  ) 
|__/|__/\__,_/_/_/____/  
The lightweight framework for web-like apps
(c) Lea Anthony 2019-present
*/
/* jshint esversion: 6 */

/**
 * Place the window in the center of the screen
 *
 * @export
 */
export function Center() {
	window.wails.Window.Center();
}

/**
 * Set the Size of the window
 *
 * @export
 * @param {number} width
 * @param {number} height
 */
export function SetSize(width, height) {
	window.wails.Window.SetSize(width, height);
}

/**
 * Set the Position of the window
 *
 * @export
 * @param {number} x
 * @param {number} y
 */
export function SetPosition(x, y) {
	window.wails.Window.SetPosition(x, y);
}

/**
 * Hide the Window
 *
 * @export
 */
export function Hide() {
	window.wails.Window.Hide();
}

/**
 * Show the Window
 *
 * @export
 */
export function Show() {
	window.wails.Window.Show();
}

/**
 * Maximise the Window
 *
 * @export
 */
export function Maximise() {
	window.wails.Window.Maximise()
}

/**
 * Unmaximise the Window
 *
 * @export
 */
export function Unmaximise() {
	window.wails.Window.Unmaximise()
}

/**
 * Minimise the Window
 *
 * @export
 */
export function Minimise() {
	window.wails.Window.Minimise();
}

/**
 * Unminimise the Window
 *
 * @export
 */
export function Unminimise() {
	window.wails.Window.Unminimise();
}

/**
 * Close the Window
 *
 * @export
 */
export function Close() {
	window.wails.Window.Close();
}

File: v3/pkg/application/window.go
package application

import (
	"github.com/wailsapp/wails/v3/pkg/events"
)

type Callback interface {
	CallError(callID string, result string, isJSON bool)
	CallResponse(callID string, result string)
	DialogError(dialogID string, result string)
	DialogResponse(dialogID string, result string, isJSON bool)
}

type Window interface {
	Callback
	Center()
	Close()
	DisableSizeConstraints()
	DispatchWailsEvent(event *CustomEvent)
	EmitEvent(name string, data ...any)
	EnableSizeConstraints()
	Error(message string, args ...any)
	ExecJS(js string)
	Focus()
	ForceReload()
	Fullscreen() Window
	GetBorderSizes() *LRTB
	GetScreen() (*Screen, error)
	GetZoom() float64
	HandleDragAndDropMessage(filenames []string)
	HandleMessage(message string)
	HandleWindowEvent(id uint)
	Height() int
	Hide() Window
	HideMenuBar()
	ID() uint
	Info(message string, args ...any)
	IsFocused() bool
	IsFullscreen() bool
	IsIgnoreMouseEvents() bool
	IsMaximised() bool
	IsMinimised() bool
	HandleKeyEvent(acceleratorString string)
	Maximise() Window
	Minimise() Window
	Name() string
	OnWindowEvent(eventType events.WindowEventType, callback func(event *WindowEvent)) func()
	OpenContextMenu(data *ContextMenuData)
	Position() (int, int)
	RelativePosition() (int, int)
	Reload()
	Resizable() bool
	Restore()
	Run()
	SetPosition(x, y int)
	SetAlwaysOnTop(b bool) Window
	SetBackgroundColour(colour RGBA) Window
	SetFrameless(frameless bool) Window
	SetHTML(html string) Window
	SetMinimiseButtonState(state ButtonState) Window
	SetMaximiseButtonState(state ButtonState) Window
	SetCloseButtonState(state ButtonState) Window
	SetMaxSize(maxWidth, maxHeight int) Window
	SetMinSize(minWidth, minHeight int) Window
	SetRelativePosition(x, y int) Window
	SetResizable(b bool) Window
	SetIgnoreMouseEvents(ignore bool) Window
	SetSize(width, height int) Window
	SetTitle(title string) Window
	SetURL(s string) Window
	SetZoom(magnification float64) Window
	Show() Window
	ShowMenuBar()
	Size() (width int, height int)
	OpenDevTools()
	ToggleFullscreen()
	ToggleMaximise()
	ToggleMenuBar()
	UnFullscreen()
	UnMaximise()
	UnMinimise()
	Width() int
	Zoom()
	ZoomIn()
	ZoomOut()
	ZoomReset() Window
}
